pkg/util/grpc/grpc.go

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright 2024 The Karmada Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package grpc
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"os"
+	"time"
+
+	"google.golang.org/grpc"
+	grpccredentials "google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+// Config the config of GRPC connection.
+type Config struct {
```

Comment:
Seems this config can be used by both the server and client side, right?
Shall we split it to two, like ServerConfig and ClientConfig? So that we can have more clear fields and without redundant.

Reply:
Some of the fields in config are actually identical on both the client and server side, such as certfile and keyfile, and splitting them into two configs may result in redundant fields. Besides, a filed like `ServerPort` is a config for server side, but both used by server and client side, making it a challenge to place this field.
Take the tls.config for example, it's a structure used to configure a TLS client or server. I think we can learn from its field definitions and comments to further improve our grpc config. What do you think?